extensions/blocks/revue/edit.js

```diff
@@ -0,0 +1,178 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { BlockControls, BlockIcon, InspectorControls } from '@wordpress/block-editor';
+import {
+	Button,
+	ToggleControl,
+	ExternalLink,
+	IconButton,
+	PanelBody,
+	Placeholder,
+	TextControl,
+	Toolbar,
+} from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import defaultAttributes from './attributes';
+import ButtonPreview from './button-preview';
+import JetpackFieldLabel from '../contact-form/components/jetpack-field-label';
+import icon from './icon';
+import { getValidatedAttributes } from '../../shared/get-validated-attributes';
+import { isAtomicSite, isSimpleSite } from '../../shared/site-type-utils';
+import './editor.scss';
+
+export default function RevueEdit( props ) {
+	const { attributes, className, setAttributes } = props;
+
+	useEffect( () => {
+		const validatedAttributes = getValidatedAttributes( defaultAttributes, attributes );
+		if ( ! isEqual( validatedAttributes, attributes ) ) {
+			setAttributes( validatedAttributes );
+		}
+	}, [ attributes ] );
+
+	const {
+		revueUsername,
+		emailLabel,
+		emailPlaceholder,
+		firstNameLabel,
+		firstNamePlaceholder,
+		firstNameShow,
+		lastNameLabel,
+		lastNamePlaceholder,
+		lastNameShow,
+	} = attributes;
+
+	const [ username, setUsername ] = useState( '' );
+
+	useEffect( () => {
+		if ( ! username && revueUsername ) {
+			setUsername( revueUsername );
+		}
+	}, [] );
+
+	const saveUsername = event => {
+		event.preventDefault();
+		setAttributes( { revueUsername: username } );
```

Comment:
Should we add a .trim() to username here? currently if a user copies and pastes some spaces along with their username by mistake they get a 404 from revue when trying to submit the form.